v2/internal/identity/credential_provider.go

```diff
@@ -34,6 +34,17 @@ const (
 	FederatedTokenFilePath = "/var/run/secrets/tokens/azure-identity"
 )
 
+type IdentityAuthModeOption string
+
+const (
+	podIdentity      IdentityAuthModeOption = "pod"
+	workloadIdentity IdentityAuthModeOption = "workload"
+
+	// IdentityAuthMode enum is used to determine if we're using Pod Identity or Workload Identity
+	//authentication for namespace and per-resource scoped credentials
+	IdentityAuthMode = "IDENTITY_AUTH_MODE"
```

Comment:
Wondering if it would make more sense to call this:
`AUTH_MODE` and have the values be `workloadidentity`, `podidentity`, or (possibly in the future) `sp` or `cert`? Calling it `IDENTITY_AUTH_MODE` might limit us?